ui/appui-react/src/appui-react/statusfields/message-center/MessageCenterField.tsx

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Notification
+ */
+import * as React from "react";
+import {
+  Button,
+  NotificationMarker,
+  Popover,
+  Tabs,
+} from "@itwin/itwinui-react";
+
+import { SvgChat } from "@itwin/itwinui-icons-react";
+import { Icon } from "@itwin/core-react";
+import { UiFramework } from "../../UiFramework";
+import { OutputMessagePriority } from "@itwin/core-frontend";
+import { MessageCenterMessage } from "./MessageCenterMessage";
+import { MessageManager } from "../../messages/MessageManager";
+import { TitleBar } from "../../layout/footer/dialog/TitleBar";
+
+import type { NotifyMessageDetailsType } from "../../messages/ReactNotifyMessageDetails";
+import "./MessageCenterField.scss";
+
+/**
+ * Type for Status state to satisfy NotificationMarker type checking
+ */
+type NotificationMarkerStatus = "primary" | "negative" | "positive";
+
+/** Message Center Field React component.
+ * @public
+ */
+export function MessageCenterField() {
+  const [messages, setMessages] = React.useState(MessageManager.messages);
+  const [notify, setNotify] = React.useState(false);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [status, setStatus] =
+    React.useState<NotificationMarkerStatus>("primary");
+
+  const indicatorRef = React.useRef<React.RefObject>({ current: null });
+  const title = UiFramework.translate("messageCenter.messages");
+
+  const handleOpenChange = (isOpenState: boolean) => {
+    setNotify(false);
+    setIsOpen(isOpenState);
+  };
+
+  const isProblemStatus = (message: NotifyMessageDetailsType): boolean => {
+    // See priority values in DgnPlatform defined in NotificationManager
+    return (
+      message.priority === OutputMessagePriority.Error ||
+      message.priority === OutputMessagePriority.Fatal
+    );
+  };
+
+  const determineStatus = () => {
+    const message = [...MessageManager.messages].pop();
+
+    if (message) {
+      if (isProblemStatus(message)) {
+        return setStatus("negative");
+      } else if (message.priority === OutputMessagePriority.Success) {
+        return setStatus("positive");
+      } else {
+        return setStatus("primary");
+      }
+    }
+    return;
+  };
+
+  React.useEffect(() => {
+    return MessageManager.onOpenMessageCenterEvent.addListener(() => {
+      handleOpenChange(true);
+    });
+  });
```

Comment:
I don't know if it's intentional but these `useEffect`s are being called on every single render because they don't have a [dependency array](https://react.dev/learn/synchronizing-with-effects#step-2-specify-the-effect-dependencies)